contracts/SimpleVaultMigrator.sol

```diff
@@ -0,0 +1,264 @@
+        uint256 migratedAmount
+    );
+
+    /* ========== MODIFIERS ========== */
+    modifier onlyGovernance {
+        require(msg.sender == governance, "Sender must be governance");
+        _;
+    }
+
+    modifier onlyPendingGovernance {
+        require(
+            msg.sender == pendingGovernance,
+            "Sender must be pending governance"
+        );
+        _;
+    }
+
+    modifier checkVaults(address vaultFrom, address vaultTo) {
+        require(
+            IVaultAPI(vaultFrom).token() == IVaultAPI(vaultTo).token(),
+            "Vaults must have the same token"
+        );
+        require(
+            IRegistry(registry).latestVault(IVaultAPI(vaultFrom).token()) ==
+                vaultTo,
+            "Target vault should be the latest for token"
+        );
+        _;
+    }
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+
+    /**
+     * @notice Migrate all of our vault shares to the newest vault version.
+     * @dev Throws if the vaults do not have the same want, or if our
+     *  target is not the newest version
+     * @param vaultFrom The old vault we are migrating from.
+     * @param vaultTo The new vault we are migrating to.
+     */
+    function migrateAll(address vaultFrom, address vaultTo) external {
+        uint256 shares = IVaultAPI(vaultFrom).balanceOf(msg.sender);
+
+        _migrate(vaultFrom, vaultTo, shares);
+    }
+
+    /**
+     * @notice Migrate a specific amount of our vault shares to the newest vault version.
+     * @dev Throws if the vaults do not have the same want, or if our
+     *  target is not the newest version
+     * @param vaultFrom The old vault we are migrating from.
+     * @param vaultTo The new vault we are migrating to.
+     * @param shares The number of shares to migrate.
+     */
+    function migrateShares(
+        address vaultFrom,
+        address vaultTo,
+        uint256 shares
+    ) external {
+        _migrate(vaultFrom, vaultTo, shares);
+    }
+
+    function _migrate(
+        address vaultFrom,
+        address vaultTo,
+        uint256 shares
+    ) internal checkVaults(vaultFrom, vaultTo) {
+        // Transfer in vaultFrom shares
+        IVaultAPI vf = IVaultAPI(vaultFrom);
+
+        uint256 preBalanceVaultFrom = vf.balanceOf(address(this));
+
+        vf.safeTransferFrom(msg.sender, address(this), shares);
+
+        uint256 balanceVaultFrom =
+            vf.balanceOf(address(this)) - preBalanceVaultFrom;
+
+        // Withdraw token from vaultFrom
+        IERC20 token = IERC20(vf.token());
+
+        uint256 preBalanceToken = token.balanceOf(address(this));
+
+        vf.withdraw(balanceVaultFrom);
+
+        uint256 balanceToken = token.balanceOf(address(this)) - preBalanceToken;
+
+        // Deposit new vault
+        token.safeIncreaseAllowance(vaultTo, balanceToken);
+
+        IVaultAPI(vaultTo).deposit(balanceToken, msg.sender);
+        emit SuccessfulMigration(msg.sender, vaultFrom, vaultTo, shares);
```

Comment:
What would it be the idea/goal for the last param in the event? afaik, the shares that the user has in the `vaultFrom` might be not equal to the `vaultTo`.
It is okay as it is but if we want to track the shares in the `vaultTo` specifically, I think we should use (or maybe add) the value returned in the `deposit` function (for the `vaultTo`).

Reply:
Sure, tbh we don't really need to have the shares, I think it's most important just to have that x user migrated from y vault to z vault, that's why I indexed those :)

Reply:
> 



> Sure, tbh we don't really need to have the shares, I think it's most important just to have that x user migrated from y vault to z vault, that's why I indexed those :)
> 
> but the logic for including `shares` was that we care far more about how many shares are _leaving_ the old vault, in case we ever wanted to track how much had migrated over or something like that. I don't see a reason why we would care about tracking vault shares created in the new vault

Perfect, thanks for the details. It makes sense to me.